apps/signal/43-signal-input/src/app/utils.ts

```diff
@@ -0,0 +1,19 @@
+import { Category } from './models';
+
+export const ageToCategory = (age: number): Category => {
+  if (age < 0 || !Number.isInteger(age)) {
+    throw new Error('Age must be a positive integer');
+  }
+
+  const categoryRanges: { [key: number]: Category } = {
+    10: 'Youth',
+    18: 'Junior',
+    35: 'Open',
+  };
```

Comment:
Why not putting this const outside of the scope of the function, so you don't recreate it at each call.